conans/client/cmd/uploader.py

```diff
@@ -82,27 +81,28 @@ def _upload(self, conan_file, conan_ref, packages_ids, retry, retry_wait,
                 integrity_check, policy, remote_name, recorder):
         """Uploads the recipes and binaries identified by conan_ref"""
 
-        defined_remote = self._registry.refs.get(conan_ref)
+        default_remote = self._registry.remotes.default
+        cur_recipe_remote = self._registry.refs.get(conan_ref)
         if remote_name:  # If remote_name is given, use it
-            upload_remote = self._registry.remotes.get(remote_name)
-        elif defined_remote:  # Else, if the package had defined a remote, use it
-            upload_remote = defined_remote
-        else:  # Or use the default otherwise
-            upload_remote = self._registry.remotes.default
+            recipe_remote = self._registry.remotes.get(remote_name)
+        else:
+            recipe_remote = cur_recipe_remote or default_remote
 
         conanfile_path = self._client_cache.conanfile(conan_ref)
+        # FIXME: I think it makes no sense to specify a remote to "pre_upload"
+        # FIXME: because the recipe can have one and the package a different one
```

Comment:
It depends on what will be the final model. If one upload of a package can only upload to a given remote, or is able to upload different binaries to different remotes, in the same command. To discuss.